src/lib/builders/tags-input/index.ts

```diff
@@ -22,15 +31,34 @@ export type CreateTagsInputArgs = {
 	// Whether the tags should be unique
 	unique?: boolean;
 	// What to do on blur
-	// blur?: 'nothing' | 'add' | 'clear';
+	blur?: 'nothing' | 'add' | 'clear';
+	// Whether the tag should be added on paste
+	addOnPaste?: boolean;
+	// Maximum number of tags (undefined || 0 or lower === no max)
+	maxTags?: number;
+	// Allowed tags
+	allowedTags?: string[];
+	// Denied tags
+	deniedTags?: string[];
+	// A custom validator function. A response of true means the validator passed
+	validator?: (tag: string) => boolean;
```

Comment:
Instead of having a validator, we should probably expose a hiddenInput that holds the value of the tagsInput. Then you can use HTML5 validation, JS validation with something like superforms, etc, otherwise it'll probably be weird to interface Tags Input validation with other libs.

Reply:
There is the $value store that holds the value of tags input, so validation can be done on that. However I’m not sure how you would block adding the tag then if the validation is done external to the builder?

Reply:
I still think we should have the add function! You could block it through there, correct?

Reply:
Ok so remove the validator? You can use the add to validate and if you aren’t happy with the tag then you can throw an error or return a reject, both of which will block the add.

Happy with this? Can also monitor the $value store and do html validation there but it wouldn’t block the add unless we monitor for an attribute of some sort? HTML validation could set an attribute and we could look for this before adding and block? Or are you happy with just doing it through the add?

Reply:
Yeah, I'd remove the validator and keep the add function.